overlord/ifacestate/handlers.go

```diff
@@ -380,12 +392,45 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return fmt.Errorf("internal error: cannot find base declaration: %v", err)
 	}
 
+	var attributes interfaces.InterfaceAttrs
+	// get attributes set by interface hooks and validate plug/slot
+	if attributes, err = getTaskHookAttributes(task); err == nil {
+		if slot.Interface != plug.Interface {
+			return fmt.Errorf(`cannot connect plug "%s:%s" (interface %q) to "%s:%s" (interface %q)`,
+				plug.Snap.Name(), plug.Name, plug.Interface, slot.Snap.Name(), slot.Name, slot.Interface)
+		}
+		iface := m.repo.Interface(slot.Interface)
+		if iface == nil {
+			return fmt.Errorf("internal error: cannot find interface: %s", slot.Interface)
+		}
+		type validatePlug interface {
+			ValidatePlug(plug *interfaces.Plug, plugAttrs map[string]interface{}) error
+		}
+		type validateSlot interface {
+			ValidateSlot(slot *interfaces.Slot, slotAttrs map[string]interface{}) error
+		}
+		if validate, ok := iface.(validatePlug); ok {
```

Comment:
it's strange to do this here, it seems repo should grow some ValidateConnection or something like that